### ISSUE TITLE
Keyboard is getting closed while focusing the textinput for the first time inside the intermediate tabs [Android only].

### ISSUE THREAD

**Issue by rifad4u** [react-navigation/react-navigation#11128]
### Current behavior

Keyboard is getting closed while focusing the textinput for the first time inside the intermediate tabs.

### Expected behavior

Keyboard shouldn’t close , should stay as opened.

### Reproduction

https://snack.expo.dev/Tv9MitB0j

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [ ] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [X] @react-navigation/​material-top-tabs
- [ ] @react-navigation/​stack
- [ ] @react-navigation/​native-stack
- [X] react-native-tab-view
- [ ] flipper-plugin-react-navigation

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |  "^6.1.1"
| @react-navigation/material-top-tabs    |  "^6.5.1"
| react-native-safe-area-context         |  "^4.4.1"
| react-native-screens                   |  "^3.18.2"
| react-native-tab-view                  |  "^3.3.4"
| react-native-pager-view                |  "^6.1.2"
| react-native                           |  "0.70.6"
| node                                   |  "14.21.1"
| npm or yarn                            | "6.14.17"

[untitled.webm](https://user-images.githubusercontent.com/37132763/210059104-39d329e6-60f6-47ad-82ea-4e3db35c22a2.webm)

**Comment by MichaelRay23** [comment]
Just set the `android:windowSoftInputMode="stateAlwaysHidden|adjustPan"`
in the MainActivity.java

**Comment by rifad4u** [comment]
> Just set the `android:windowSoftInputMode="stateAlwaysHidden|adjustPan"` in the MainActivity.java

I can’t do this , bcz i need to use keyboardavoiding view , if i set this method then keyboard avoiding doesn’t work.

**Comment by BrechtBD** [comment]
same issue here

```
"react-native": "0.68.5"
"react-native-tab-view": "3.3.4"
"react-native-pager-view": "6.1.2"
```

**Comment by mehdinourollah** [comment]
> There is a hack for this defect by disabling animation on Android:
> 
> ```
> const TopTab = createMaterialTopTabNavigator()
> 
> <TopTab.Navigator
> screenOptions={{
>   ...otherProps,
>   animationEnabled: Platform.OS === 'ios',
> }}
> ```
> 
> It's not a proper solution though. Let's hope it will be fixed in the future.

Works like a charm ! Thanks

**Comment by trungtruc91** [comment]
> 

work for me, thanks

**Comment by okwasniewski** [comment]
Hey, 

The bug does not lie in tab-view or pager-view but in [ViewPager2](https://developer.android.com/jetpack/androidx/releases/viewpager2) (the native android library that pager-view uses). There are issues in the Google Issue tracker around keyboard handling: https://issuetracker.google.com/issues/149937409  So right now there is not much we can do about this bug except use a workaround. 

A possible solution is to set `android:windowSoftInputMode="adjustPan"` in `AndroidManifest.xml`. 

For expo, inside `app.json`: 
```json
"android": {
    "softwareKeyboardLayoutMode" : "pan"
},
```

Let me know if this fixes your issue.

**Comment by santoshpatelnita** [comment]
Still got same issue

**Comment by trantus** [comment]
Same issue, happens for me only in case if you set more than 2 routes (in my case 3)
And only happens on the second route (index: 1)
`animationEnabled={Platform.OS === 'android'}` - helped

**Comment by saquibhayat26** [comment]
I was able to identify the exact cause of the issue—it's occurring for tabs other than the first and last ones. The problem is related to the animationEnabled prop, which is set to true by default, causing the keyboard to close on the first interaction. To resolve this, you need to set animationEnabled to false.

The issue you're encountering is caused by the default behavior of the animationEnabled prop in react-native-tab-view. When animationEnabled is set to true (which is the default), it adds a sliding animation when navigating between tabs. This animation can interfere with the keyboard behavior, causing it to close when switching to or interacting with certain tabs (especially those that are not the first or last).

By setting animationEnabled={false}, you disable the tab switching animation. This prevents the keyboard from closing unexpectedly when switching between tabs, providing a smoother user experience when interacting with input fields on the affected tabs.

In summary, the reason for the issue is that the animation was affecting the keyboard behavior, and disabling it resolves the problem.

I am currently using the following versions:

"react-native-tab-view": "^3.5.2"
"react-native": "0.72.5".
Here's the updated code snippet:

<TabView
  navigationState={{index, routes}}
  renderScene={renderScene}
  onIndexChange={setIndex}
  swipeEnabled={swipeEnabled}
  initialLayout={initialLayout}
  renderTabBar={hasMoreTab ? renderTabBarMore : renderTabBar}
  scrollEnabled={true}
  animationEnabled={false}  <--------this line
/>

**Comment by Isaccobosio** [comment]
> I was able to identify the exact cause of the issue—it's occurring for tabs other than the first and last ones. The problem is related to the animationEnabled prop, which is set to true by default, causing the keyboard to close on the first interaction. To resolve this, you need to set animationEnabled to false.
> 
> **The issue you're encountering is caused by the default behavior of the animationEnabled prop in react-native-tab-view. When animationEnabled is set to true (which is the default), it adds a sliding animation when navigating between tabs. This animation can interfere with the keyboard behavior, causing it to close when switching to or interacting with certain tabs (especially those that are not the first or last).**
> 
> By setting animationEnabled={false}, you disable the tab switching animation. This prevents the keyboard from closing unexpectedly when switching between tabs, providing a smoother user experience when interacting with input fields on the affected tabs.
> 
> In summary, the reason for the issue is that the animation was affecting the keyboard behavior, and disabling it resolves the problem.
> 
> **I am currently using the following versions:
> 
> "react-native-tab-view": "^3.5.2" "react-native": "0.72.5".** Here's the updated code snippet:
> 
> <TabView navigationState={{index, routes}} renderScene={renderScene} onIndexChange={setIndex} swipeEnabled={swipeEnabled} initialLayout={initialLayout} renderTabBar={hasMoreTab ? renderTabBarMore : renderTabBar} scrollEnabled={true} **animationEnabled={false}** <--------this line />

Is it solved with version 4.x and react navigation 7.x?

**Comment by gunwooa** [comment]
> For anyone still having an issue, try `keyboardDismissMode="none"`
> 
> ```
> <TabView
>  keyboardDismissMode="none"
>  navigationState={{ index, routes }}
>  onIndexChange={handleTabChange}
>  renderTabBar={renderTabBar}
>  renderScene={renderScene}
> />
> ```
> 
> related: [satya164/react-native-tab-view#1181 (comment)](https://github.com/satya164/react-native-tab-view/issues/1181#issuecomment-846596541)

thank you bro..

**Comment by patel-malav** [comment]
> > I was able to identify the exact cause of the issue—it's occurring for tabs other than the first and last ones. The problem is related to the animationEnabled prop, which is set to true by default, causing the keyboard to close on the first interaction. To resolve this, you need to set animationEnabled to false.
> > 
> > **The issue you're encountering is caused by the default behavior of the animationEnabled prop in react-native-tab-view. When animationEnabled is set to true (which is the default), it adds a sliding animation when navigating between tabs. This animation can interfere with the keyboard behavior, causing it to close when switching to or interacting with certain tabs (especially those that are not the first or last).**
> > 
> > By setting animationEnabled={false}, you disable the tab switching animation. This prevents the keyboard from closing unexpectedly when switching between tabs, providing a smoother user experience when interacting with input fields on the affected tabs.
> > 
> > In summary, the reason for the issue is that the animation was affecting the keyboard behavior, and disabling it resolves the problem.
> > 
> > **I am currently using the following versions:
> > 
> > "react-native-tab-view": "^3.5.2" "react-native": "0.72.5".** Here's the updated code snippet:
> > 
> > <TabView navigationState={{index, routes}} renderScene={renderScene} onIndexChange={setIndex} swipeEnabled={swipeEnabled} initialLayout={initialLayout} renderTabBar={hasMoreTab ? renderTabBarMore : renderTabBar} scrollEnabled={true} **animationEnabled={false}** <--------this line />
> 
> Is it solved with version 4.x and react navigation 7.x?

Nope
